src/eager.ts

```diff
@@ -0,0 +1,29 @@
+import { isAsyncIterable } from './utils/iterator'
+import { LazyIterable } from './shared-types'
+
+export function eager<T>(size: number) {
+  return function eagerFn(data: LazyIterable<T>) {
+    if (data == null) return
+
+    // TODO: Do we need to deal with this?
+    if (isAsyncIterable(data) || data instanceof Promise) return data
+
+    return {
+      *[Symbol.iterator]() {
+        const buffer = []
+        const iterator = data[Symbol.iterator]()
+        let state = { more: true }
+
+        while (state.more) {
+          for (let i = 0; i < size; i++) {
+            const { value, done } = iterator.next()
+            if (done) state.more = false
+            if (!done) buffer.push(value)
+          }
+
+          yield* buffer.splice(0)
+        }
```

Comment:
Regarding async iterables here, I think there's a good case to let them through and process them in the same manner!

On line `19` (when pulling out `next()`), that would return a promise if `data` is asynciterable, all that's needed to do in that case is await it, before extracting `value` and `done`. Everything else could stay the same, I think... `await` on a non-promise would just resolve right away, am I right?